### ISSUE TITLE
Identify lncRNAs in Cv RNA-seq data set

### ISSUE THREAD

**Issue by sr320** [RobertsLab/resources#1375]
Possible pipelines

https://www.sciencedirect.com/science/article/pii/S0145305X17300630
>o do this, contigs were passed through a series of filters which are aimed to identify and discard all transcripts that could potentially encode for proteins. Thus, contigs with an average coverage below 50 reads per base and with lengths above 1000 nt were removed for further analysis. Remaining contigs were BLAST against NCBI non-redundant (nr) database for nucleotides and proteins (BLASTn and BLASTx) and all hits with E values > E−10 were considered as significant blast hit and therefore removed. Finally, remaining contigs with open reading frames above 200 nt and with significant coding potential estimated through the Coding Potential Assessment Tool (CPAT) (Wang et al., 2013) were also removed. The remaining contigs were considered as putative lncRNAs and used for further analysis.

also 

https://www.nature.com/articles/s41598-018-19950-6

>We developed a stringent filtering pipeline designed to remove transcripts with evidence for protein-coding potential based on current approaches: (i) Filter out single-exon transcripts nearest distance ≤500 bp with other transcripts, which might be extended exons of annotated protein-coding genes2. (ii) Remove transcripts with short lengths (<200 bp). (iii) Select single-exon transcripts with FPKM ≥ 2 and multiple-exon transcripts with FPKM ≥ 0.5. (iv) Filter out transcripts that belong to tRNA, rRNA, snoRNA, snRNA, pre-miRNA, and pseudogenes by cuffcompare. (v) The remaining transcripts were blasted with known mRNA and completed the preliminary screening. (vi) Classify candidate lncRNAs into three subtypes (lincRNA, intronic lncRNA, and antisense lncRNA) using information of calss_code of cuffcompare. (vii) The tools of CPC (Coding Potential Calculator), CPAT (Coding-Potential Assessment Tool), Pfamscan were used to detect putative protein encoding transcripts, potential lncRNA transcripts were retained which are not detected in any tool. (viii) Select putative lncRNAs which can be detected in at least three libraries.

**Comment by sr320** [comment]
see also https://github.com/RobertsLab/resources/issues/1302#issuecomment-931565411

**Comment by kubu4** [comment]
Well, after discussion at lab meeting today, it looks like this has already been done!

![image](https://user-images.githubusercontent.com/4514104/154145050-fb4442ba-ae16-4131-a112-efaf378d625f.png)

Sorry for not thinking of this earlier. I made the assumption that someone had looked at this already, since you provided the papers describing different approaches to IDing lncRNAs.

I'll work on getting a dedicated GFF and BED files.

**Comment by kubu4** [comment]
@sr320 Quick question on what you'd like from this.

tl;dr - easy to get gene level info, but challenging to get exon level info. Do you want exon level info, too?

---

NCBI provides a [features table file](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002/022/765/GCF_002022765.2_C_virginica-3.0/GCF_002022765.2_C_virginica-3.0_feature_table.txt.gz) (gzipped text file) that looks like this:

```
grep "^ncRNA" GCF_002022765.2_C_virginica-3.0_feature_table.txt | head

ncRNA	lncRNA	GCF_002022765.2	Primary Assembly	chromosome	1	NC_035780.1	13578	14594	+	XR_002636969.1			uncharacterized LOC111116054	LOC111116054	111116054		118	118	
ncRNA	lncRNA	GCF_002022765.2	Primary Assembly	chromosome	1	NC_035780.1	169468	170178	-	XR_002635081.1			uncharacterized LOC111105702	LOC111105702	111105702		199	199	
ncRNA	lncRNA	GCF_002022765.2	Primary Assembly	chromosome	1	NC_035780.1	900326	903430	+	XR_002636046.1			uncharacterized LOC111111519	LOC111111519	111111519		1645	1645
```

NCBI also provides a [GTF file](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002/022/765/GCF_002022765.2_C_virginica-3.0/GCF_002022765.2_C_virginica-3.0_genomic.gtf.gz) (gzipped GTF) that looks like this:

```
head GCF_002022765.2_C_virginica-3.0_genomic.gtf

#gtf-version 2.2
#!genome-build C_virginica-3.0
#!genome-build-accession NCBI_Assembly:GCF_002022765.2
#!annotation-source NCBI Crassostrea virginica Annotation Release 100
NC_035780.1	Gnomon	gene	13578	14594	.	+	.	gene_id "LOC111116054"; db_xref "GeneID:111116054"; gbkey "Gene"; gene "LOC111116054"; gene_biotype "lncRNA"; 
NC_035780.1	Gnomon	exon	13578	13603	.	+	.	gene_id "LOC111116054"; transcript_id "XR_002636969.1"; db_xref "GeneID:111116054"; gbkey "ncRNA"; gene "LOC111116054"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 1 sample with support for all annotated introns"; product "uncharacterized LOC111116054"; exon_number "1"; 
NC_035780.1	Gnomon	exon	14237	14290	.	+	.	gene_id "LOC111116054"; transcript_id "XR_002636969.1"; db_xref "GeneID:111116054"; gbkey "ncRNA"; gene "LOC111116054"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 1 sample with support for all annotated introns"; product "uncharacterized LOC111116054"; exon_number "2"; 
NC_035780.1	Gnomon	exon	14557	14594	.	+	.	gene_id "LOC111116054"; transcript_id "XR_002636969.1"; db_xref "GeneID:111116054"; gbkey "ncRNA"; gene "LOC111116054"; model_evidence "Supporting evidence includes similarity to: 100% coverage of the annotated genomic feature by RNAseq alignments, including 1 sample with support for all annotated introns"; product "uncharacterized LOC111116054"; exon_number "3"; 
```

The features table makes it relatively easy and straightforward to extract info to create BED, GTF, and FastA files for all lncRNAs. _But_, this only grabs the lncRNAs at the gene level (i.e. exons have been spliced together). 

The GTF _has_ the exon info for each lncRNA, but parsing these out to create BED, GTF, and FastA files becomes significantly more challenging.

I'll go ahead and get the gene-level lncRNA files produced, as it's very straightforward (and, I'm almost done).

Do you want me to also work on getting files made which includes info at the exon level, too?

**Comment by kubu4** [comment]
Guess I'm tired today. Have this easily figured out for exon retrieval. You can ignore my previous post. Sorry!

**Comment by kubu4** [comment]
- [GCF_002022765.2_C_virginica-3.0_lncRNA.bed](https://gannet.fish.washington.edu/Atumefaciens/20220217-cvir-lncRNA_subsetting/GCF_002022765.2_C_virginica-3.0_lncRNA.bed) (652K)

  - MD5: `db0047b39b04bdb8d25dd35005117513`

- [GCF_002022765.2_C_virginica-3.0_lncRNA.fa](https://gannet.fish.washington.edu/Atumefaciens/20220217-cvir-lncRNA_subsetting/GCF_002022765.2_C_virginica-3.0_lncRNA.fa) (5.5M)

  - MD5: `4610b254477051dac32e4bac3684651c

- [GCF_002022765.2_C_virginica-3.0_lncRNA.fa.fai](https://gannet.fish.washington.edu/Atumefaciens/20220217-cvir-lncRNA_subsetting/GCF_002022765.2_C_virginica-3.0_lncRNA.fa.fai) (176K)

  - MD5: `3dc5789c9184eea9060b977dd97c3f1a`

- [GCF_002022765.2_C_virginica-3.0_lncRNA.gff](https://gannet.fish.washington.edu/Atumefaciens/20220217-cvir-lncRNA_subsetting/GCF_002022765.2_C_virginica-3.0_lncRNA.gff) (5.7M)

  - MD5: `6347a08c2deddff0fc27c1da82c02d40`

- [GCF_002022765.2_C_virginica-3.0_lncRNA.gtf](https://gannet.fish.washington.edu/Atumefaciens/20220217-cvir-lncRNA_subsetting/GCF_002022765.2_C_virginica-3.0_lncRNA.gtf) (2.9M)

  - MD5: `1a18aa39d6055d348b7971a979a737b1`

**Comment by kubu4** [comment]
Re-opened (and deleted previous post with links to files). Comparing output files with different approach to make sure they're correct. Sorry for the tease!

**Comment by kubu4** [comment]
Corrected files:

- [GCF_002022765.2_C_virginica-3.0_lncRNA.bed](https://gannet.fish.washington.edu/Atumefaciens/20220217-cvir-lncRNA_subsetting/GCF_002022765.2_C_virginica-3.0_lncRNA.bed) (640K)

  - MD5: `28de37c9ee1308ac1175397d16b3aafe`

- [GCF_002022765.2_C_virginica-3.0_lncRNA.fa](https://gannet.fish.washington.edu/Atumefaciens/20220217-cvir-lncRNA_subsetting/GCF_002022765.2_C_virginica-3.0_lncRNA.fa) (25M)

  - MD5: `7fac9e7191915f763cc7f5d22838ac25`

- [GCF_002022765.2_C_virginica-3.0_lncRNA.fa.fai](https://gannet.fish.washington.edu/Atumefaciens/20220217-cvir-lncRNA_subsetting/GCF_002022765.2_C_virginica-3.0_lncRNA.fa.fai) (180K)

  - MD5: `1b43db284950abc07afb5f50164fb264`

- [GCF_002022765.2_C_virginica-3.0_lncRNA.gff](https://gannet.fish.washington.edu/Atumefaciens/20220217-cvir-lncRNA_subsetting/GCF_002022765.2_C_virginica-3.0_lncRNA.gff) (2.1M)

  - MD5: `00755b8c80166cdec94b09f231ef440a`

- [GCF_002022765.2_C_virginica-3.0_lncRNA.gtf](https://gannet.fish.washington.edu/Atumefaciens/20220217-cvir-lncRNA_subsetting/GCF_002022765.2_C_virginica-3.0_lncRNA.gtf) (1.3M)

  - MD5: `dedab056acd679cf4eab83629882ee10`
